app/views/unattended/provisioning_templates/registration/global_registration.erb

```diff
@@ -86,8 +86,23 @@ EOF
 elif [ -f /etc/debian_version ]; then
   <%= save_to_file('/etc/apt/sources.list.d/foreman_registration.list', @repo) %>
 <% if @repo_gpg_key_url.present? -%>
+<%
+  <% # Ubuntu 16.04 does not support GPG public keys in "/etc/apt/trusted.gpg.d/" -%>
```

Comment:
https://manpages.debian.org/testing/apt/apt-key.8.en.html states:
> Make sure to use the "asc" extension for ASCII armored keys and the "gpg" extension for the binary OpenPGP format (also known as "GPG key public ring"). The binary OpenPGP format works for all apt versions, while the ASCII armored format works for apt version >= 1.4.

So I think you may be hitting this.